### ISSUE TITLE
Error when using impersonated service account

### ISSUE THREAD

**Issue by imathews** [GoogleCloudPlatform/gcsfuse#647]
I'm attempting to use an impersonated service account to authorize GCS fuse in a development environment (`gcloud auth application-default login --impersonate-service-account=<service_account_name>`). 

However, I'm getting the following error when running gcsfuse:

```sh
daemonize.Run: readFromProcess: sub-process: mountWithArgs: getConn: 
DefaultTokenSource: google: error getting credentials using GOOGLE_APPLICATION_CREDENTIALS environment variable: 
unknown credential type: "impersonated_service_account"
```

I noticed that this case _should_ be [handled in the source code](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/vendor/golang.org/x/oauth2/google/google.go#L188), but for whatever reason the service account isn't being recognized as an `impersonatedServiceAccount`.

If it's helpful, the credentials being referenced look like:
```sh
$ cat $GOOGLE_APPLICATION_CREDENTIALS
{
  "delegates": [],
  "service_account_impersonation_url": "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/<redacted>",
  "source_credentials": {
    "client_id": "<redacted>",
    "client_secret": "<redacted>",
    "refresh_token": "<redacted>",
    "type": "authorized_user"
  },
  "type": "impersonated_service_account"
}
```

**Comment by avidullu** [comment]
Could you please let us know the version of GCSFuse that you're running?
Can you repro this on the latest 0.40 version?

**Comment by imathews** [comment]
@avidullu thanks so much! The issue was that I was on an older version, sorry for the trouble.

For others who might come across this: my distro is Debian bullseye, and I ran into the same issue as described in #553. I set the release to `jessie` during installation, which installed an older version (0.21 I believe). Updating the hardcoded release to `stretch` or `buster` properly installed v0.40, which resolved this issue.
